src/config/mcp-config.js

```diff
@@ -115,33 +115,106 @@ class MCPConfigManager {
   }
 
   /**
-   * Build complete MCP configuration for Amazon Q CLI
+   * Get allowed tools string for Claude CLI (with wildcards and specific tool names)
+   */
+  getAllowedToolsForClaude() {
+    const workingDir = process.env.GITHUB_WORKSPACE || process.cwd();
+
+    const allowedTools = [
+      // File operations (restricted to target repository)
+      `Read(${workingDir}/**)`,
+      `Edit(${workingDir}/**)`,
+      `MultiEdit(${workingDir}/**)`,
+      `Glob(${workingDir}/**)`,
+      `Grep(${workingDir}/**)`,
+
+      // Git operations
+      "Bash(git status:*)",
+      "Bash(git log:*)",
+      "Bash(git diff:*)",
+      "Bash(git show:*)",
+      "Bash(git checkout:*)",
+      "Bash(git branch:*)",
+
+      // System commands (restricted to working directory)
+      `Bash(ls:${workingDir}/**)`,
+      `Bash(find:${workingDir}/**)`,
+      `Bash(cat:${workingDir}/**)`,
+      `Bash(head:${workingDir}/**)`,
+      `Bash(tail:${workingDir}/**)`,
+      `Bash(wc:${workingDir}/**)`,
+
+      // GitHub MCP tools
+      "mcp__github__*",
+      ...this.getGitHubToolsList()
+    ];
+
+    // Add AWS MCP tools if credentials are available
+    if (this.hasAWSCredentials()) {
+      allowedTools.push(
+        "mcp__*",
+        "mcp__awslabs_cloudwatch-appsignals-mcp-server__*",
+        ...this.getApplicationSignalsToolsList()
+      );
+
+      // Add CloudWatch MCP tools if enabled
+      if (this.hasCloudWatchAccess()) {
+        allowedTools.push(
+          "mcp__awslabs_cloudwatch-mcp-server__*",
+          ...this.getCloudWatchToolsList()
+        );
+      }
+    }
+
+    return allowedTools.join(',');
+  }
+
+  /**
+   * Build complete MCP configuration for specified CLI type
+   * @param {string} cliType - 'claude' or 'amazonq'
    * @returns {object} MCP configuration object
    */
-  buildMCPConfig() {
+  buildMCPConfig(cliType = 'amazonq') {
     const config = { mcpServers: {} };
 
     // Add AWS CloudWatch Application Signals MCP server if credentials available
     if (this.hasAWSCredentials()) {
       const applicationSignalsConfig = this.getApplicationSignalsServerConfig();
 
-      // Amazon Q CLI format
-      config.mcpServers["applicationsignals"] = {
-        ...applicationSignalsConfig,
-        autoApprove: this.getApplicationSignalsToolsList(),
-        disabled: false
-      };
+      if (cliType === 'claude') {
+        // Claude CLI format
+        config.mcpServers["awslabs.cloudwatch-appsignals-mcp-server"] = {
+          ...applicationSignalsConfig,
+          env: this.getAWSEnvVars()
```

Comment:
this will override the `MCP_RUN_FROM` env var set in getApplicationSignalsConfig.